tests/e2e/utils/pages/product-feed.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Internal dependencies
+ */
+import { LOAD_STATE } from '../constants';
+import MockRequests from '../mock-requests';
+
+/**
+ * ProductFeed page object class.
+ */
+export default class ProductFeedPage extends MockRequests {
+	/**
+	 * @param {import('@playwright/test').Page} page
+	 */
+	constructor( page ) {
+		super( page );
+		this.page = page;
+	}
+
+	/**
+	 * Go to the product feed page.
+	 *
+	 * @return {Promise<void>}
+	 */
+	async goto() {
+		await this.page.goto(
+			'/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed',
+			{ waitUntil: LOAD_STATE.DOM_CONTENT_LOADED }
+		);
+	}
+
+	/**
+	 * Mock all requests related to external accounts such as Merchant Center, Google, etc.
+	 *
+	 * @return {Promise<void>}
+	 */
+	async mockRequests() {
+		await Promise.all( [
+			this.fulfillMCReview( {
+				cooldown: 0,
+				issues: [],
+				reviewEligibleRegions: [],
+				status: 'ONBOARDING',
+			} ),
+
+			this.fulfillAccountIssuesRequest( {
+				issues: [],
+				page: 1,
+				total: 0,
+				loading: false,
+			} ),
+
+			this.fulfillJetPackConnection( {
+				active: 'yes',
+				owner: 'yes',
+				displayName: 'John',
+				email: 'john@email.com',
+			} ),
+
+			this.mockGoogleConnected(),
+
+			this.fulfillAdsConnection( {
+				id: 1111111,
+				currency: 'USD',
+				symbol: '$',
+				status: 'connected',
+				step: '',
+			} ),
+		] );
+	}
+
+	/**
+	 * Change the value of adsSetupComplete to `true` after it is discovered.
+	 *
+	 * @return {Promise<void>}
+	 */
+	async mockAdsSetupComplete() {
```

Comment:
@eason9487 `useAdsCampaigns` mainly relies on the value of  `adsSetupComplete` to make a campaign query. I tried using `mockCampaignCreationAndAdsSetupCompletion` on this page, but it didn't work here. Finally, I went ahead modifying the global as soon as it was available.

Let me know if there is a better approach here.

Reply:
Done this way.